boards/arm/mimxrt595_evk/board.c

```diff
@@ -4,8 +4,178 @@
  */
 
 #include <zephyr/init.h>
+#include <zephyr/drivers/regulator/consumer.h>
+#include <zephyr/dt-bindings/regulator/pca9420_i2c.h>
+#include <zephyr/pm/policy.h>
 #include "fsl_power.h"
 
+#if CONFIG_REGULATOR_PMIC && CONFIG_PM
+const struct device *sw1 = DEVICE_DT_GET(DT_NODELABEL(pca9420_sw1));
+const struct device *sw2 = DEVICE_DT_GET(DT_NODELABEL(pca9420_sw2));
+const struct device *ldo1 = DEVICE_DT_GET(DT_NODELABEL(pca9420_ldo1));
+const struct device *ldo2 = DEVICE_DT_GET(DT_NODELABEL(pca9420_ldo2));
+
+/*
+ * This function sets up the PMIC operation in different low power modes.
+ */
+static int setup_pmic_modes(void)
+{
+	int ret = 0;
+
+	/*
+	 * Initial mode is set to Mode0 through the device tree property.
+	 * Also the regulators are configured to be enabled in the initial mode.
+	 */
+
+	/* SW1 setting in Run mode: Enabled @1V */
+	ret = regulator_set_voltage(sw1, 1000000, 1000000);
```

Comment:
wouldn't be better having these voltage values defined in DT ?

Reply:
The voltage levels are pre-defined in the SoC datasheet. I do not see a need for the user to specify these values.

Reply:
@ceolin after discussing with the internal team it seem like having this come from DT is a good idea. 
Do you have a suggestion on where to place these PMIC properties?

Reply:
I have updated the implementation to use DT